### PR TITLE
[Unblock clean CI] Remove the CI step to restore vcpkg from the cache and remove the analyze flag for MSVC

### PR DESCRIPTION
**Pull request by ahsonkhan**
Testing out some stopgap solution to https://github.com/Azure/azure-sdk-for-c/issues/1142 by removing the restore from cache step.

Also, remove the analyze flag for MSVC to get clean CI to test out a stopgap which unblocks existing PRs while investigation is pending for https://github.com/Azure/azure-sdk-for-c/issues/1137

### CODE REVIEW

File: eng/pipelines/templates/steps/vcpkg.yml [hunk 1]
```diff
@@ -88,6 +70,10 @@ steps:
         git clone https://github.com/Microsoft/vcpkg.git
         cd vcpkg
         git rev-parse --verify HEAD
+
+        # Make sure to fetch all the tags to avoid failures on checkout.
+        git fetch --all --tags
```
Comment (ahsonkhan) [inline eng/pipelines/templates/steps/vcpkg.yml:75]: @danieljurek Turns out we were failing to checkout the vcpkg tag causing changes from latest master to come in, causing out-of-sync state on the machine.

`git checkout tags/2020.06` was failing with:
> pathspec '…' did not match any file(s) known to git

Fixed by fetching all the tags first.

**Comment by momuno** [comment]
Why did the analyze flag create the problem?  Was it newly added, or did something underneath change?

**Comment by ahsonkhan** [comment]
> Why did the analyze flag create a problem? Was it newly added, or did something underneath change?

The MSVC compiler on the build machines upgraded which brought in new (possibly false positive) analysis capabilities.

See https://github.com/Azure/azure-sdk-for-c/issues/1137#issuecomment-681078133

Nothing was changed recently from the SDK code perspective that caused these new warnings.

File: eng/pipelines/templates/steps/vcpkg.yml (hunk 1 above)
Comment (weshaggard) [reply eng/pipelines/templates/steps/vcpkg.yml:75]: Yes this is an interesting problem when ever the tag isn't rooted by the default branch you are forced to also fetch the tags.

Comment (weshaggard) [reply eng/pipelines/templates/steps/vcpkg.yml:75]: FYI: "--all" isn't needed for our case as it just says to fetch from all remotes which we only have one of.

Comment (ahsonkhan) [reply eng/pipelines/templates/steps/vcpkg.yml:75]: Btw, is there a way for us to capture failures from running the various git commands and fail the leg? I am not  sure for how long this `git checkout` command was failing for, silently.

Comment (weshaggard) [reply eng/pipelines/templates/steps/vcpkg.yml:75]: Yes but we have to look at the exit codes explicitly.  See https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/scripts/git-branch-push.ps1 as examples. If you don't explicitly look at the exit codes the script will simply use the exit code for the last command you ran. That is why it is become a little tricky to have a bunch of commands that might fail in an inline script.

Comment (ahsonkhan) [reply eng/pipelines/templates/steps/vcpkg.yml:75]: That's a pretty good self-documenting script with good error messages being logged. Makes it quite easy to figure out what step failed and why just by looking at the logs.

**Pull request merged**